examples/nametoken/src/lib.rs

```diff
@@ -0,0 +1,1719 @@
+//! A NameToken smart contract example using the Concordium Token Standard
+//! CIS2.
+//!
+//! # Description
+//! A contract that allows for registering and managing names. A name can be
+//! used as a verifiable human-readable name for an address. They also can have
+//! some data attached, like website URLs, accounts in various systems, etc.
+//! Only account addresses can be owners. The ownership can be transferred to
+//! another account address.
+//!
+//! # Operations
+//! The contract allows for
+//!  - registering new names;
+//!  - updating data associated with a name;
+//!  - renewing a name;
+//!  - viewing information about a name (owner and data) To register, renew and
+//! update one has to pay a fee. Registration succeeds if either name is not
+//! yet registered, or it is registered but expired. In this case, ownership is
+//! transferred to the new owner and the expiration date is updated as for the
+//! "fresh" registration. Updating data and renewing is only possible if the
+//! name is not expired.
+//!
+//! # Token
+//! NameToken is essentially an NFT. Therefore we expose it as an instance of
+//! CIS-2 standard. This allows trading of domain names on NFT auctions. The
+//! contract is based on the NFT example with modifications required for the
+//! name management. Registering a fresh name is effectively minting. It
+//! generates log events accordingly. However, taking over an expired name is
+//! not considered minting. Ownership can be transferred by the owner or by an
+//! operator. Owners are accounts only, but operators can be any address. Token
+//! ownership is determined by the expiration date. Expired name tokens are not
+//! burned, they are considered as not owned, that is the balance is 0 for the
+//! address it was initially registered.
+//!
+//! Note: token ids are hashed names. Words "name" and "token id" are used
+//! interchangeably
+//!
+//! # Misc
+//! This example demontrates how to use crypto primitives (hashing) and lazy
+//! loaded data.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use concordium_cis2::*;
+use concordium_std::*;
+
+/// The baseurl for the token metadata, gets appended with the token ID as hex
+/// encoding before emitted in the TokenMetadata event.
+const TOKEN_METADATA_BASE_URL: &str = "https://some.example/nametoken/";
+
+/// List of supported standards by this contract address.
+const SUPPORTS_STANDARDS: [StandardIdentifier<'static>; 2] =
+    [CIS0_STANDARD_IDENTIFIER, CIS2_STANDARD_IDENTIFIER];
+
+// Fees
+
+// Registration fee in CCD
+const REGISTRACTION_FEE: Amount = Amount::from_ccd(70);
+
+// Data update fee in CCD
+const UPDATE_FEE: Amount = Amount::from_ccd(7);
+
+// Extension fee in CCD
```

Comment:
Suggestion:
The comment could use “renewal fee” instead of "extension fee" to be consistent.